### PR TITLE
Add active field to person sentences

### PR DESCRIPTION
**Pull request by popey2700**
#### Work done
- Added isActive to Hmpps-Integration-Api Sentence model from both Nomis and nDelius
- Created a mapping function for Nomis sentenceStatus String to Boolean

#### Tests
Does this PR include tests for the changes introduced?

- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/nomis/Sentence.kt [hunk 1]
```diff
@@ -5,8 +5,17 @@ import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Sentence as Integ
...
 data class Sentence(
   val sentenceDate: LocalDate?,
+  val sentenceStatus: String?,
 ) {
   fun toSentence(): IntegrationApiSentence = IntegrationApiSentence(
     dateOfSentencing = this.sentenceDate,
+    isActive = toIsActive(this.sentenceStatus),
   )
 }
+private fun toIsActive(sentenceStatus: String?): Boolean? {
```
Comment (luke-a-williams) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/nomis/Sentence.kt:15]: This function name hurts my OCD. The question "isActive?", returning a boolean of true of false in certain cases of sentenceStatus, is already sufficient for this as a function name, no?
toIsActive implies we are changing the state of something, but we aren't doing this in my head?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/SentencesControllerTest.kt [hunk 2]
```diff
@@ -70,10 +72,13 @@ internal class SentencesControllerTest(
           """
           [
             {
-              "dateOfSentencing": "1990-01-01"
+              "dateOfSentencing": "1990-01-01",
+              "isActive": true
```
Comment (luke-a-williams) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/SentencesControllerTest.kt:76]: Should we also have a test for `"isActive": null`? Why exclude it?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/ndelius/GetSentencesForPersonTest.kt [hunk 3]
```diff
@@ -65,9 +66,11 @@ class GetSentencesForPersonTest(
           listOf(
             generateTestSentence(
               dateOfSentencing = LocalDate.parse("2009-07-07"),
+              isActive = false,
             ),
             generateTestSentence(
               dateOfSentencing = LocalDate.parse("2009-09-01"),
+              isActive = false,
```
Comment (luke-a-williams) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/ndelius/GetSentencesForPersonTest.kt:73]: Is there something significant about the date being different whilst isActive remains the same? Otherwise should this be again testing all of the possible values (true, false, null)?

**Review by luke-a-williams — CHANGES_REQUESTED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/nomis/PersonSentencesTest.kt [hunk 4]
```diff
@@ -9,19 +9,47 @@ import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.nomis.Sentence as
...
+          sentenceStatus = "A",
         )
 
         val integrationApiSentence = nomisSentence.toSentence()
+
+        integrationApiSentence.dateOfSentencing.shouldBe(nomisSentence.sentenceDate)
+        integrationApiSentence.isActive.shouldBe(true)
+      }
+
+      it("toIsActive correctly maps 'I' to false") {
```
Comment (luke-a-williams) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/nomis/PersonSentencesTest.kt:25]: You changed the function name so it needs to be changed in the test name too

**Review by luke-a-williams — CHANGES_REQUESTED** [review]

**Review by luke-a-williams — APPROVED** [review]
gtm
